services/httpd/handler.go

```diff
@@ -993,8 +1012,366 @@ func fixLiterals(node influxql.Node) {
 	}
 }
 
+type RetentionRule struct {
+	Type                      string `json:"type"`
+	EverySeconds              int64  `json:"everySeconds"`
+	ShardGroupDurationSeconds int64  `json:"shardGroupDurationSeconds"`
+}
+
+// BucketsBody and RetentionRule should match the 2.0 API definition.
```

Comment:
Did we consider just importing the swagger / openapi types for this? I'm not sure they would work out of the box but would be nice to move in that direction if possible.

Reply:
I did not open that can of worms. I'd start with that in OSS 2.X, and only do it here if we continue to see the V2 API getting  a lot of use in 1.10.